### PR TITLE
update wix package install

### PR DESCRIPTION
**Pull request by tgauth**
# PR Summary
- install WiX from Nuget feed and move install to build script from pipeline yaml

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [X] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/wix.psm1 [hunk 1]
```diff
@@ -4,14 +4,18 @@ Import-Module "$PSScriptRoot\dockerInstall.psm1"
...
 
     $targetRoot = $arm64 ? "${env:ProgramFiles(x86)}\Arm Support WiX Toolset xcopy" : "${env:ProgramFiles(x86)}\WiX Toolset xcopy"
+    # cleanup previous install
+    if(Test-Path $targetRoot) {
+        Remove-Item $targetRoot -Recurse -Force
+    }
     $binPath = Join-Path -Path $targetRoot -ChildPath 'bin'
-    Write-Verbose "Expanding $zipPath to $binPath ..." -Verbose
-    Expand-Archive -Path $zipPath -DestinationPath $binPath -Force
+    Register-PSRepository -Name NuGetGallery -SourceLocation https://api.nuget.org/v3/index.json
```
Comment (TravisEz13) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/wix.psm1:17]: We need to use https://pkgs.dev.azure.com/powershell/PowerShell/_packaging/powershell/nuget/v3/index.json or https://pkgs.dev.azure.com/powershell/PowerShell/_packaging/powershell/nuget/v2

also, we should have csproj so compliance systems knows that we use the package.

**Review by TravisEz13 — APPROVED** [review]

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/wix.psm1 [hunk 2]
```diff
@@ -4,14 +4,19 @@ Import-Module "$PSScriptRoot\dockerInstall.psm1"
 
 # Install using Wix Zip because the MSI requires an older version of dotnet
```
Comment (TravisEz13) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/wix.psm1:5]: can we move both these files to `tools/wix`

File: build.psm1 [hunk 3]
```diff
@@ -778,7 +778,7 @@ function Restore-PSPackage
 
     if (-not $ProjectDirs)
     {
-        $ProjectDirs = @($Options.Top, "$PSScriptRoot/src/TypeCatalogGen", "$PSScriptRoot/src/ResGen", "$PSScriptRoot/src/Modules")
+        $ProjectDirs = @($Options.Top, "$PSScriptRoot/src/TypeCatalogGen", "$PSScriptRoot/src/ResGen", "$PSScriptRoot/src/Modules", "$PSScriptRoot/tools/releaseBuild/Images/microsoft_powershell_windowsservercore")
```
Comment (TravisEz13) [inline build.psm1:781]: still old path

**Review by TravisEz13 — APPROVED** [review]

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
Need support

**Pull request merged**